next.config.js

```diff
@@ -17,11 +17,61 @@ const basePath = '/developer';
 
 const accountPortalUrl = `${process.env.ACCOUNT_PORTAL_URL}/account/login`;
 
+const buildImagePatterns = (hostsString = '') => {
+    const items = hostsString.split(';');
+
+    if (items.length === 0) {
+        return [];
+    }
+
+    return items.reduce((acc, item) => {
+        const match =
+            /^(https?):\/\/([a-zA-Z0-9\*\.\-]+)((?:\/[a-zA-Z0-9\*\-]+)+)?/.exec(
+                item.trim()
+            );
+
+        if (match && match[2]) {
+            return [
+                ...acc,
+                {
+                    protocol: match[1],
+                    hostname: match[2],
+                    ...(match[3] ? { pathname: match[3] } : {}),
+                },
+            ];
+        } else {
+            return acc;
+        }
+    }, []);
+};
+
+/*
+    Set the NEXT_IMAGE_HOSTS env variable to a semicolon-delimited list of hosts to add extra image hosts
+    e.g. export NEXT_IMAGE_HOSTS="https://google.com/;http://**.test.com/test?abc=123" will generate the entries
+    [{ 'protocol': 'https', 'hostname': 'google.com' }, { 'protocol': 'http', 'hostname': '**.test.com', 'pathname': '/test' }]
+*/
+const IMAGE_PATTERNS = buildImagePatterns(process.env.NEXT_IMAGE_HOSTS);
```

Comment:
Cool! Thanks for adding this. I think `NEXT_IMAGE_HOSTS` would need to be set in the drone pipeline, as we set our other env vars at build time. I guess we can use a drone secret for now (though one day I will want to find a way to not use drone for this). Let me know if you need help setting this up. https://github.com/mongodb/devcenter/blob/6a60b966891605493c045d893b3c7b97992df675/.drone.yml#L15-L37